### PR TITLE
Small refactor for data fetching and add support for accepting admin value from API endpoint

### PR DESCRIPTION
**Pull request by hswolff**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/mongo-nav/src/MongoNav.tsx [hunk 1]
```diff
@@ -18,12 +18,104 @@ import {
...
+  | 'activeProjectId'
+  | 'loadData'
+  | 'dataFixtures'
+  | 'onSuccess'
+  | 'onError'
+> & {
+  hosts: Required<NonNullable<MongoNavInterface['hosts']>>;
+};
+
+function useMongoNavData({
```
Comment (DesignerDave) [inline packages/mongo-nav/src/MongoNav.tsx:41]: Really really like breaking this out. Much more easy to reason about independent from the rest of the component.

File: packages/mongo-nav/src/MongoNav.tsx [hunk 2]
```diff
@@ -18,12 +18,104 @@ import {
...
+          'Content-Type': 'application/json',
+        },
+      });
+    }
+
+    return fetch(endpointURI, configObject);
+  }
+
+  function getDataFixtures() {
+    return new Promise(resolve => {
```
Comment (DesignerDave) [inline packages/mongo-nav/src/MongoNav.tsx:76]: I believe if you do `return new Promise<DataInterface>(resolve => {` here, you can avoid typecasting, and simplify line 104 to:
```js
getDataFixtures().then(setData);
```

File: packages/mongo-nav/src/MongoNav.tsx [hunk 3]
```diff
@@ -87,13 +179,24 @@ function MongoNav({
...
+    mode,
+    activeOrgId,
+    activeProjectId,
+    loadData,
+    dataFixtures,
+    onSuccess,
+    onError,
+  });
+
+  const admin = (adminProp ?? data?.account?.admin) === true;
```
Comment (DesignerDave) [inline packages/mongo-nav/src/MongoNav.tsx:199]: 😁

Comment (hswolff) [reply packages/mongo-nav/src/MongoNav.tsx:199]: lol. The 10 minute long line of code.

File: packages/mongo-nav/src/MongoNav.tsx (hunk 2 above)
Comment (hswolff) [reply packages/mongo-nav/src/MongoNav.tsx:76]: Nice catch! This was copy pasta but I'll fix this up!

**Pull request merged**